app/src/main/java/ch/epfl/polybazaar/FillListingActivity.java

```diff
@@ -216,9 +222,25 @@ private void submit() {
                 }
             };
             String category = spinnerList.get(spinnerList.size()-1).getSelectedItem().toString();
-            Listing newListing = new Listing(titleSelector.getText().toString(), descriptionSelector.getText().toString(), priceSelector.getText().toString(), "test.user@epfl.ch", stringImage, category);
+            Listing newListing = new Listing(titleSelector.getText().toString(), descriptionSelector.getText().toString(), priceSelector.getText().toString(), "test.user@epfl.ch", "", category);
             LiteListing newLiteListing = new LiteListing(newListingID, titleSelector.getText().toString(), priceSelector.getText().toString(), category);
             storeListing(newListing, newListingID, successCallback);
+
+            //store images (current has a ref to the next)
+            if(listStingImage.size() > 0) {
+                String currentId = newListingID;
+                String nextId;
+                for(int i = 0; i < (listStingImage.size() - 1); i++) {
+                    nextId = randomUUID().toString();
+                    ListingImage newListingImage = new ListingImage(listStingImage.get(i), nextId);
+                    storeListingImage(newListingImage, currentId, result -> {});
```

Comment:
Wouldn't it make sense to perform some kind of check in the callback and log if the image was successfully send (using Log.d()). Maybe even go further and do some error management, i.e. don't have this function call fail silently but maybe send a toast.

Reply:
Ok, I will log the result and add a Toast to inform the user if an error has occurred.